### ISSUE TITLE
Config file generated by --generate-toml-config option produces errors when later used

### ISSUE THREAD

**Issue by mrechte** [PyCQA/pylint#7654]
### Bug description

pylint --generate-toml-config >~/.config/pylintrc

The generated file seems to be unusable by pylint.

### Command used

```shell
pylint mymodule
```

### Pylint output

```shell
usage: pylint [options]
pylint: error: argument --py-version: "3.10" has an invalid format, should be a version string. E.g., '3.8'
```

### Expected behavior

no error

### Pylint version

```shell
pylint --version
pylint 2.15.4
astroid 2.12.12
Python 3.10.8 (main, Oct 11 2022, 20:04:56) [GCC 12.2.0]
```

### OS / Environment

Archlinux

**Comment by nelfin** [comment]
It's interpreting the TOML file as an rcfile, probably because it's called `pylintrc` no? See this behaviour:

```bash
$ cat pyproject.toml 
[tool.pylint."messages control"]
disable = ["all"]
$ pylint foo.py 

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 6.67/10, +3.33)

$ mv pyproject.toml pylintrc
$ pylint foo.py
************* Module .../pylintrc
pylintrc:1:0: W0012: Unknown option value for '--disable', expected a valid pylint message and got '["all"]' (unknown-option-value)
************* Module foo
foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```

I don't know what the expected behaviour _should_ be here, but I kind of think it's sensible to expect that `pylintrc` should be rc-formatted and `pyproject.toml` (`pylint.toml`?) should be TOML formatted.

**Comment by Pierre-Sassoulas** [comment]
>  (pylint.toml?)

Let's not create yet another place where the conf can be, there's already ``setup.cfg``, ``pylintrc``, ``.pylintrc``, ``/etc/conf/.pylintrc`` (?), ``/etc/conf/pylintrc`` (?), ``pyproject.toml`` and wherever you want with ``rcfile`` 😄

We could handle mixup between ini/toml more gracefully, but is it worth the additional bug it's going to create when we have a edge case in our edge case (say the file look like valid ini and parse like a valid ini but it's in fact a toml) ?

**Comment by nelfin** [comment]
What about instead of guessing the format (especially since TOML and INI are deliberately similar), updating the examples in the documentation to nudge the behaviour that rcfile = `pylintrc` and toml-config = `pyproject.toml`? Something like:

```reST
You can generate a sample configuration file with `--generate-toml-config` or `--generate-rcfile`. Every
option present on the command line before this will be included in the file.

For example:
    pylint --class-rgx='[A-Z][a-z]+' >> pylintrc
or
    pylint --disable=bare-except,invalid-name --generate-toml-config >> pyproject.toml
```

**Comment by DanielNoord** [comment]
Is there any argument against using `pyproject.toml`? Even if just for `pylintrc`?

**Comment by Pierre-Sassoulas** [comment]
I separate the two config file because pylint is pretty verbose and configurable and often default values are not the one I would use so it's flooding the ``pyproject.toml`` with a wall of options. (better default values will probably have to wait for 4.0 too or we'll never release 3.0)

**Comment by mrechte** [comment]
> What about instead of guessing the format (especially since TOML and INI are deliberately similar), updating the examples in the documentation to nudge the behaviour that rcfile = `pylintrc` and toml-config = `pyproject.toml`? Something like:
> 
> ```
> You can generate a sample configuration file with `--generate-toml-config` or `--generate-rcfile`. Every
> option present on the command line before this will be included in the file.
> 
> For example:
>     pylint --class-rgx='[A-Z][a-z]+' --generate-rcfile >> pylintrc
> or
>     pylint --disable=bare-except,invalid-name --generate-toml-config >> pyproject.toml
> ```

Being new to pylint, I did not pay attention there was 2 config file formats. Yes, man page could just be updated to make it clearer.
Thanks
